### PR TITLE
Add CI tests for an external driver

### PR DESCRIPTION
**Pull request by dgageot**
Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -31,23 +27,12 @@ test:
     - gvm use stable && go version
     - gvm use stable && make build:
         pwd: $BASE_STABLE
+    - gvm use stable && GO15VENDOREXPERIMENT=1 go get github.com/dgageot/docker-machine-driver-none-external
```
Comment (nathanleclaire) [inline circle.yml]: I know this is convenient, but can we find a way to do this without referencing a repository outside of this one? (e.g. using `cp -r` and `go install` from a subfolder of this repo)  If this repository needs to be updated it should be doable by anyone who has access to the Machine repo.

Comment (dgageot) [inline circle.yml]: I fear that having a folder in the `docker/machine` repo that contains a copy of some packages/dependencies of `docker/machine` itself would be a bit surprising. By having an external repo, we also protect ourselves from unwanted modifications during a refactoring. Could we have a repo under `docker` organization?

Comment (nathanleclaire) [inline circle.yml]: Yes, I think a repo under `docker` organization might be the best approach.

**Comment by jeanlaurent** [comment]
LGTM

**Comment by nathanleclaire** [comment]
Sent an email asking to get a new repo created for this functionality, after we accomplish that I'm good to go on this.  Thanks!

File: circle.yml (hunk 1 above)
Comment (nathanleclaire) [inline circle.yml]: @dgageot https://github.com/docker/docker-machine-driver-ci-test (for reference)

**Comment by nathanleclaire** [comment]
LGTM after repo bump :+1:

**Comment by jeanlaurent** [comment]
@nathanleclaire I believe the repo has been up, I let you merge if you feel like it.

**Pull request merged**